js/tests/enhancedDate.spec.js

```diff
@@ -0,0 +1,69 @@
+// describing spec (what's being tested)
+describe("enhancedDate", function () {
+
+	// Test each method within object
+
+	// setDate
+	describe("setDate", function () {
+
+		it("Return same date object as date object given.", function () {
+			var newDate = new Date("Dec 12, 2015");
+			var newDateString = newDate.toString();
+			var setDateResult = enhancedDate.setDate(newDate);
+			var setDateString = setDateResult.toString();
+
+			expect(newDateString).to.equal(setDateString);
+		});
+
+		it("Give unix time then get date object back that represents that same time.", function () {
+			var dateNumber = 1432406605;
+			var setDateResult = enhancedDate.setDate(dateNumber);
+			var setDateNumber = setDateResult.valueOf();
+
+			expect(dateNumber).to.equal(setDateNumber);
```

Comment:
Can you tell me if I’m on the right track with testing booleans for isToday?


// isToday
describe("isToday", function () {

   it("Test should return true.", function () {
      var setToday = true;

      expect(enhancedDate.isToday()).to.equal(setToday);
   });

   it("Test should return false.", function () {
      var setToday = false;

      expect(enhancedDate.isToday()).to.equal(setToday);
   });
});



> On May 24, 2015, at 3:14 PM, jamdon <notifications@github.com> wrote:
> 
> In js/tests/enhancedDate.spec.js:
> 
> > +
> > +		it("Return same date object as date object given.", function () {
> > +			var newDate = new Date("Dec 12, 2015");
> > +			var newDateString = newDate.toString();
> > +			var setDateResult = enhancedDate.setDate(newDate);
> > +			var setDateString = setDateResult.toString();
> > +
> > +			expect(newDateString).to.equal(setDateString);
> > +		});
> > +
> > +		it("Give unix time then get date object back that represents that same time.", function () {
> > +			var dateNumber = 1432406605;
> > +			var setDateResult = enhancedDate.setDate(dateNumber);
> > +			var setDateNumber = setDateResult.valueOf();
> > +
> > +			expect(dateNumber).to.equal(setDateNumber);
> 
> Yes, that's correct. The setDate method is saying it will do something, and we're validating that it will do what it says it will. In this case, if we provide setDate a millisecond value, it should be creating a new Date object from that value and returning it. We validate this is what happens by creating a new native Date object using that value and validating the two create identical objects.
> 
> —
> Reply to this email directly or view it on GitHub.
>